tests/sps/formats/sps_xml/test_ref.py

```diff
@@ -89,3 +91,227 @@ def test_build_ref_sub_elements_None(self):
+    def test_build_ref_pub_id(self):
+        data = {
+            "ref-id": "B1",
+            "publication-type": "journal",
+            "pub-ids": [
+                {
+                    "pub-id-type": "pmid",
+                    "text": "15867408"
+                }
+            ]
+        }
+        expected_xml_str = (
+            '<ref id="B1">'
+            '<element-citation publication-type="journal" />'
+            '<pub-id pub-id-type="pmid">15867408</pub-id>'
+            '</ref>'
+        )
+        ref_elem = build_ref(data)
+        generated_xml_str = ET.tostring(ref_elem, encoding="unicode", method="xml")
+        self.assertEqual(generated_xml_str.strip(), expected_xml_str.strip())
+
+    def test_build_ref_pub_id_None(self):
+        data = {
+            "ref-id": "B1",
+            "publication-type": "journal",
+            "pub-ids": None
+        }
+        expected_xml_str = (
+            '<ref id="B1">'
+            '<element-citation publication-type="journal" />'
+            '</ref>'
+        )
+        ref_elem = build_ref(data)
+        generated_xml_str = ET.tostring(ref_elem, encoding="unicode", method="xml")
+        self.assertEqual(generated_xml_str.strip(), expected_xml_str.strip())
+
+    def test_build_ref_pub_id_attribs_None(self):
+        data = {
+            "ref-id": "B1",
+            "publication-type": "journal",
+            "pub-ids": [
+                {
+                    "pub-id-type": None,
+                    "text": "15867408"
+                }
+            ]
+        }
+        with self.assertRaises(ValueError) as e:
+            build_ref(data)
+        self.assertEqual(str(e.exception), "pub-id-type is required")
+
+    def test_build_ref_ext_link_text_None(self):
+        data = {
+            "ref-id": "B1",
+            "publication-type": "journal",
+            "pub-ids": [
+                {
+                    "pub-id-type": "pmid",
+                    "text": None
+                }
+            ]
+        }
+        expected_xml_str = (
+            '<ref id="B1">'
+            '<element-citation publication-type="journal" />'
+            '</ref>'
+        )
+        ref_elem = build_ref(data)
+        generated_xml_str = ET.tostring(ref_elem, encoding="unicode", method="xml")
+        self.assertEqual(generated_xml_str.strip(), expected_xml_str.strip())
+
+
+class TestBuildDateInCitation(unittest.TestCase):
+    def test_build_ref_date_in_citation(self):
+        data = {
+            "ref-id": "B1",
+            "publication-type": "journal",
+            "dates-in-citation": [
+                {
+                    "content-type": "updated",
+                    "text": "2006 Jul 20"
+                }
+            ]
+        }
+        expected_xml_str = (
+            '<ref id="B1">'
+            '<element-citation publication-type="journal" />'
+            '<date-in-citation content-type="updated">2006 Jul 20</date-in-citation>'
```

Comment:
@Rossi-Luciano date-in-citation tem que estar dentro de element-citation

Reply:
OK, @robertatakenaka